src/Storages/Distributed/DistributedSettings.h

```diff
@@ -26,6 +26,7 @@ class ASTStorage;
     M(UInt64, monitor_split_batch_on_failure, 0, "Default - distributed_directory_monitor_split_batch_on_failure", 0) \
     M(Milliseconds, monitor_sleep_time_ms, 0, "Default - distributed_directory_monitor_sleep_time_ms", 0) \
     M(Milliseconds, monitor_max_sleep_time_ms, 0, "Default - distributed_directory_monitor_max_sleep_time_ms", 0) \
+    M(UInt64, monitor_max_retries, 0, "Default - distributed_directory_file_send_max_retries", 0) \
```

Comment:
So this new setting is a table level setting? I think it would be better if this setting is a global setting along with other settings like `distributed_directory_monitor_max_sleep_time_ms`. 
In this way, operators can easily change this setting in global for all distributed tables. 

Another suggestion is it's better to make this setting to be in effect immediately once it's updated in the config.xml.

Reply:
yes, this new setting is a table level setting, I considered it mainly to be compatible with the previous case where there was no limit on retries, so I set it by table level

Reply:
Actually those monitor settings should not be per-query at all, since they are applied only once.
I've added table-level settings and later was going to deprecated them.

Reply:
> Actually those monitor settings should not be per-query at all, since they are applied only once. I've added table-level settings and later was going to deprecated them.

yes, it's actually just for Exception, so I can optimize that

Reply:
yes，points to optimize: 1. Global configuration; 2. query for exception only